### PR TITLE
fix: redirect 500 error

### LINKED ISSUE #1580: Set up next.js to do not redirect to `/500` when encountering 500 error
The `404` page is fine, as shown below:
image

But the `500` page will cause the redirection:
image
We need to show the 500 but remain on the original URL

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- For design guidelines checkout our [Figma design](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap).
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

### PR DESCRIPTION
**Pull request by lichenqin1**
# Description

[comment]: # 'Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.'

Fixes #1580

Remove redirect for 500 error and use the custom error page to return 500 error. It only works in production due to the design of Next.js, not sure if there is a better way.

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

[comment]: # 'Please include screenshots of your changes if relevant.'

|       Before        |       After        |
| :-----------------: | :----------------: |
|  |  |

# How Has This Been Tested?

- [ ] Cypress integration
- [ ] Cypress component tests
- [ ] Jest unit tests

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/models/utils.js [hunk 1]
```diff
@@ -230,10 +230,10 @@ const wrapper = (callback) => (params) =>
...
     if (e.response?.status === 404) return { notFound: true };
     return {
-      redirect: {
-        destination: '/500',
-        permanent: false,
-      },
+      // redirect: {
+      //   destination: '/500',
+      //   permanent: false,
+      // },
```
Comment (dadiorchen) [inline src/models/utils.js:236]: I think commenting out code is not a good practice, to keep the code clean,  let's simply remove it.

**Review by dadiorchen — COMMENTED** [review]
@lichenqin1 I wait for this PR for months :D thank you! please remove the comment above and I will merge it.

File: src/models/utils.js (hunk 1 above)
Comment (lichenqin1) [reply src/models/utils.js]: I removed the commented code, thanks for the suggestion.

**Comment by dadiorchen** [comment]
@lichenqin1 thank you!

**Pull request merged**